Assets/Scripts/PuzzleScript/TargetMover.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TargetMover : MonoBehaviour
+{
+    public Transform[] points; // Points Array
```

Comment:
The variables **points** and **moveSpeed** are currently exposed as public. It would be helpful to clarify in the comments why these variables need to remain public. If they are intended to be adjusted via the Unity Inspector, consider adding [SerializeField] to limit access while retaining flexibility in the Editor.